indico/web/client/js/custom_elements/ind_collapsible.js

```diff
@@ -0,0 +1,79 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import {domReady} from 'indico/utils/domstate';
+
+import './ind_collapsible.scss';
+
+customElements.define(
+  'ind-collapsible',
+  class extends HTMLElement {
+    connectedCallback() {
+      domReady.then(() => {
+        this.container = this.querySelector('[data-collapsible]');
+        this.containerStyle = window.getComputedStyle(this.container);
+
+        console.assert(
+          this.container,
+          '<ind-collapsible> must have a descendant with a data-collapsible attribute'
+        );
+        console.assert(
+          this.querySelector('button[value="toggle-collapse"]'),
+          '<ind-collapsible> must have a descendant button with value="toggle-collapse"'
+        );
+
+        this.markCollapsible();
+
+        window.addEventListener('resize', () => this.markCollapsible());
+        window.addEventListener('orientationchange', () => this.markCollapsible());
+        this.addEventListener('click', evt => {
+          if (evt.target.closest('button[value="toggle-collapse"]')) {
+            this.toggleAttribute('collapsed');
+          }
+        });
+      });
+    }
+
+    disconnectedCallback() {
+      window.removeEventListener('resize', this.markCollapsible);
+      window.removeEventListener('orientationchange', this.markCollapsible);
+    }
+
+    markCollapsible() {
+      const children = this.container.children;
+      let lastChild = children[0];
+      let linesFound = 1;
+      const cutOffLines = Number(this.getAttribute('lines'));
+
+      for (let i = 1, child; (child = children[i++]); ) {
```

Comment:
Would a normal `for...of` loop work?

Reply:
Yes, but it would be quite a bit slower.

Reply:
how often does the resize event trigger? if this triggers hundreds or thousands of times while resizing the browser window, maybe the call should be debounced so it doesn't trigger as often?

Reply:
Interesting.. then I think it'd be nice to add a comment there explaining why this type of for loop is used

Reply:
Actually, it wouldn't even work, cause starting with index 1, now that I look at it again. :))

Reply:
> how often does the resize event trigger? if this triggers hundreds or thousands of times while resizing the browser window, maybe the call should be debounced so it doesn't trigger as often?

I don't think it triggers that often. Maybe 15~20 times per second. But I think it would be useful to debounce it anyway for a smoother UX.